pandas/io/pytables.py

```diff
@@ -3294,7 +3324,11 @@ def read(
         index = self.read_index("index", start=start, stop=stop)
         values = self.read_array("values", start=start, stop=stop)
         result = Series(values, index=index, name=self.name, copy=False)
-        if using_string_dtype() and is_string_array(values, skipna=True):
+        if (
+            using_string_dtype()
+            and isinstance(values, np.ndarray)
```

Comment:
I'm assuming `read_array` can return a non-ndarray type here? Does this prevent against pyarrow arrays?

Reply:
> I'm assuming `read_array` can return a non-ndarray type here?

Correct - datetime and string currently.